concert/ext/ufo.py

```diff
@@ -682,27 +682,19 @@ def score(vector):
 
         return result
 
-    async def _copy_projections(self, producer):
-        def copy_projection(projection):
+    async def _process_projection(self, projection):
+        def copy_projection():
             self.projections[self._num_received_projections] = projection
 
-        async for projection in producer:
-            if self._num_received_projections == 0:
-                if not self.args.width:
-                    (self.args.height, self.args.width) = projection.shape
-                elif (self.args.height, self.args.width) != projection.shape:
-                    raise GeneralBackprojectManagerError('Projections have different '
-                                                         'shape from normalization images')
-                in_shape = (self.args.number,) + projection.shape
-                if (self.projections is None or in_shape != self.projections.shape
-                        or projection.dtype != self.projections.dtype):
-                    self.projections = np.empty(in_shape, dtype=projection.dtype)
+        if self._num_received_projections < self.args.number:
+            await run_in_executor(copy_projection)
+            self._num_received_projections += 1
+            async with self._producer_condition:
+                self._producer_condition.notify_all()
 
-            if self._num_received_projections < self.args.number:
-                await run_in_executor(copy_projection, projection)
-                self._num_received_projections += 1
-                async with self._producer_condition:
-                    self._producer_condition.notify_all()
+    async def _copy_projections(self, producer):
```

Comment:
I find the naming a bit confusing since `_copy_projections` calls `_process_projections` which internally calls `copy_projections`.

Reply:
Yeah, but it calls `copy_projection` defined inside the method and nothing of this is part of public API anyway.